src/SDKs/Azure.Base/data-plane/Azure.Base.Tests/BufferResponsePolicyTests.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Azure.Base.Http.Pipeline;
+using Azure.Base.Testing;
+using NUnit.Framework;
+
+namespace Azure.Base.Tests
+{
+    public class BufferResponsePolicyTests: PolicyTestBase
+    {
+        [Test]
+        public async Task ReadsEntireBodyIntoMemoryStream()
+        {
+            MockResponse mockResponse = new MockResponse(200);
+            var readTrackingStream = new ReadTrackingStream(128, int.MaxValue);
+            mockResponse.ResponseContentStream = readTrackingStream;
+
+            var mockTransport = new MockTransport(mockResponse);
+            var response = await SendGetRequest(mockTransport, BufferResponsePolicy.Singleton);
+
+            Assert.IsInstanceOf<MemoryStream>(response.ContentStream);
+            var ms = (MemoryStream)response.ContentStream;
+
+            Assert.AreEqual(128, ms.Length);
+            foreach (var b in ms.ToArray())
+            {
+                Assert.AreEqual(223, b);
```

Comment:
I'm a bit confused as to where the `223` comes from and why it is expected.  It may help to add some comments for the reader.

Reply:
I found my answer below;  to reinforce my comment there, I think that it would be really helpful to read something like `Assert.AreEqual(ReadTrackingStream.ContentByteValue);`  than to be presented with `223` with no context around it.